higher-order.js

```diff
@@ -18,13 +58,10 @@ const _ = {
             throw new TypeError('The passed mapper closure must be a valid one.');
         }
 
-        let mapped = [];
-
-        for (let i = 0; i < arr.length; i++) {
-            mapped.push(mapper(arr[i], i, arr));
-        }
-
-        return mapped;
+        return this.reduce(arr, (previous, current, index, array) => {
```

Comment:
Nice seeing you implement `map` in terms of `reduce`!

As more people do this exercise we'll talk about what arguments make sense here and why, but I understand why you're passing all that in — JavaScript's `map` does the same.

Reply:
Exactly, I'm not using the JS implementation, but I'm implementing it to be similar to the JS one.